src/components/inline_edit/inline_edit.tsx

```diff
@@ -0,0 +1,197 @@
+// import { useEuiTheme } from '../../services';
+// import { euiInlineEditStyles } from './inline_edit.styles';
+import { htmlIdGenerator } from '../../services/accessibility';
+import { useEuiI18n } from '../i18n';
+
+export const DISPLAY_TYPES = ['title', 'text'] as const;
+export type EuiInlineEditDisplayType = typeof DISPLAY_TYPES[number];
+
+interface TitleDisplayProps {
+  display: 'title';
+  size?: EuiTitleSize;
+}
+
+interface TextDisplayProps {
+  display: 'text';
+  size?: TextSize;
+}
+
+export type EuiInlineEditProps = CommonProps &
+  ExclusiveUnion<TextDisplayProps, TitleDisplayProps> & {
+    /**
+     * Display type for the text in readView
+     */
+    display: EuiInlineEditDisplayType;
+    /**
+     * Default string value for input in readView
+     */
+    defaultValue: string;
+    /**
+     * Allow users to pass in a function when the confirm button is clicked
+     */
+    onConfirm?: () => void;
+    /**
+     * Form label that appears above the form control
+     * This is required for accessibility because there is no visual label on the input
+     */
+    inputAriaLabel: String;
+    saveButtonAriaLabel?: string;
+    cancelButtonAriaLabel?: string;
+    /**
+     * Start in editView
+     */
+    startWithEditOpen?: boolean;
+  };
+
+export const EuiInlineEdit: FunctionComponent<EuiInlineEditProps> = ({
+  children,
+  className,
+  display,
+  // m is the default for both EuiTitle and EuiText
+  size = 'm',
+  defaultValue,
+  onConfirm,
+  inputAriaLabel,
+  saveButtonAriaLabel,
+  cancelButtonAriaLabel,
+  startWithEditOpen,
+  ...rest
+}) => {
+  const classes = classNames('euiInlineEdit', className);
+
+  // Styles to come later
+  /*const theme = useEuiTheme();
+  const styles = euiInlineEditStyles(theme);
+  const cssStyles = [styles.euiInlineEdit];*/
+
+  const [isInEdit, setIsInEdit] = useState(startWithEditOpen);
+  const inlineEditInputId = htmlIdGenerator('__inlineEditInput')();
+
+  /* Text Controls */
+  const [editViewValue, setEditViewValue] = useState(defaultValue);
+  const [readViewValue, setReadViewValue] = useState(defaultValue);
+
+  const editTextViewOnChange = (e: any) => {
+    setEditViewValue(e.target.value);
+  };
+
+  const saveTextEditValue = () => {
+    const input = (document.getElementById(
+      inlineEditInputId
+    ) as HTMLInputElement).value;
+
+    // If there's no text, cancel the action, reset the input text, and return to readView
+    if (input) {
+      setReadViewValue(input);
+      setIsInEdit(!isInEdit);
+      onConfirm && onConfirm();
+    } else {
+      setEditViewValue(readViewValue);
+      setIsInEdit(!isInEdit);
+    }
+  };
+
+  const defaultSaveButtonAriaLabel = useEuiI18n(
+    'euiInlineEdit.saveButtonAriaLabel',
+    'Save'
+  );
+
+  const defaultCancelButtonAriaLabel = useEuiI18n(
+    'euiInlineEdit.cancelButtonAriaLabel',
+    'Save'
+  );
+
+  const textEditViewElement = (
+    <EuiForm fullWidth>
+      <EuiFlexGroup>
```

Comment:
We can decrease the `gutterSize`:

```suggestion
      <EuiFlexGroup gutterSize="s">
```

![image](https://user-images.githubusercontent.com/2750668/220688317-6cb69cfc-3abc-4f0f-b194-86668d57ba02.png)